ui/src/app/views/workspace/component.ts

```diff
@@ -117,6 +118,11 @@ export class WorkspaceComponent implements OnInit, OnDestroy {
     const wsData: WorkspaceData = this.route.snapshot.data.workspace;
     this.workspace = wsData;
     this.accessLevel = wsData.accessLevel;
+    if (this.workspace.researchPurpose.approved !== true
```

Comment:
`if (!condA && condB)`.

Even better is to pull these out of the if block and just declare directly what they are.

```
const {approved, reviewRequested} = this.workspace.researchPurpose;
this.awaitingReview = reviewRequested && !approved;
```

Does `reviewRequested` become false when a review is finished?  This seems to indicate that `awaitingReview` would be true for workspaces that have requested a review and _not_ been approved.

Reply:
Correct, the cases that should be blocked because they are in a waiting for review status should be:
Requested review and not looked at yet.
Requested review and rejected.
---
Should pass are:
requested review and approved
Did not request review.